packages/flutter/lib/src/rendering/error.dart

```diff
@@ -50,6 +51,9 @@ class RenderErrorBox extends RenderBox {
 
   /// The message to attempt to display at paint time.
   final String message;
+  final double systemTopPadding = Zone.current['systemTopPadding'] == null
```

Comment:
Would you do that everywhere that reads from the `ui.window` static?  For example, the Zone approach would also let you test that WidgetApp correctly sets up the root MediaQueryData from the window:

https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/app.dart#L188

It would also let you test that the devicePixelRatio updates properly when the window notifies that its metrics have changed.

Another benefit of the Zone approach is that it will let you test code that someone else wrote that wasn't expecting to be tested in this way.